### PR TITLE
Add support for controllers to throw exceptions

### PR DESCRIPTION
**Pull request by rahulkavale**
- Change description
Adds support to make the generated controller methods have `throws exception` in the signature in a configurable way. The behaviour can be controlled with a configuration parameter `controllerMethodThrowsException` which takes a boolean value representing if the exception clause is to be added to the controller method. Since the behaviour can be toggled from configuration with default value as false. Hence the change is backward compatible.

- Requirement for the feature
To handle the exceptions from various places including the controller methods in a consistent way, Spring has [`@ExceptionHandler`](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/bind/annotation/ExceptionHandler.html) annotation. When using the `@ExceptionHandler`, the logic to handle a particular exception can be done at a single place avoiding duplication.
More details with example can be found at [Spring documentation](https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc#global-exception-handling)

- [x] Added Tests
- [x] Updated documentation
- [x] Backward compatible change

### CODE REVIEW

**Comment by stojsavljevic** [comment]
Why don't you use unchecked (runtime) exceptions?

File: springmvc-raml-plugin/README.md [hunk 1]
```diff
@@ -278,6 +278,11 @@ Then simply include the following code in the POM of the project you wish to gen
 ### baseUri
 (optional) Base URI for generated Spring controllers. This overrules the baseUri attribute from inside the .raml spec.
 
+### controllerMethodThrowsException
+(optional) Parameter which decides if the generated methods in controller have `throws Exception` in the signature.
+This is to use the spring ``@ExceptionHandler` for handling exceptions.
```
Comment (stojsavljevic) [inline springmvc-raml-plugin/README.md]: `@ExceptionHandler` can handle runtime exceptions too. We are adding `throws Exception` in order to support checked exceptions. The reason for that can be existing exception system of checked exceptions in an application that is starting to use the plugin.
So please remove this line.

File: springmvc-raml-plugin/README.md [hunk 2]
```diff
@@ -278,6 +278,11 @@ Then simply include the following code in the POM of the project you wish to gen
...
+When set to true, all the methods in all the controllers have the throws Exception added to the signature. Default value is false.
```
Comment (stojsavljevic) [inline springmvc-raml-plugin/README.md]: .. all the controllers will have `throws Exception` added to..

**Comment by stojsavljevic** [comment]
As I stated in the comment above, `@ExceptionHandler` can handle runtime exceptions so I'm not sure why are you referring to `@ExceptionHandler` feature.
Also, in the future please open an issue before creating a PR.

**Comment by rahulkavale** [comment]
@stojsavljevic Cool, I had already made few changes and wanted to submit it back to community. Thanks for the feedback.

**Pull request closed without merging**